curvefs/src/metaserver/inode_manager.cpp

```diff
@@ -316,22 +309,84 @@ MetaStatusCode InodeManager::GetOrModifyS3ChunkInfo(
             }
         }
         if (!duplicated) {
-            for (auto &item : s3ChunkInfoAdd) {
-                auto it = old.mutable_s3chunkinfomap()->find(item.first);
-                if (it != old.mutable_s3chunkinfomap()->end()) {
-                    MergeToS3ChunkInfoList(item.second, &(it->second));
-                } else {
-                    old.mutable_s3chunkinfomap()->insert(
-                        {item.first, item.second});
+            if (fromS3Compaction) {
```

Comment:
This logical is a little bit long, please spilt it into small functions.

Reply:
done